src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -2498,23 +2555,129 @@ def managed_db_restore(
-    kwargs['source_database_id'] = _get_managed_db_resource_id(
-        cmd.cli_ctx,
-        resource_group_name,
-        managed_instance_name,
-        database_name)
+
+    if deletion_date:
+        kwargs['restorable_dropped_database_id'] = _get_managed_dropped_db_resource_id(
+            cmd.cli_ctx,
+            resource_group_name,
+            managed_instance_name,
+            database_name,
+            deletion_date)
+    else:
+        kwargs['source_database_id'] = _get_managed_db_resource_id(
+            cmd.cli_ctx,
+            resource_group_name,
+            managed_instance_name,
+            database_name)
 
     return client.create_or_update(
         database_name=target_managed_database_name,
         managed_instance_name=target_managed_instance_name,
         resource_group_name=target_resource_group_name,
         parameters=kwargs)
 
+
+def update_short_term_retention_mi(
+        cmd,
+        client,
+        database_name,
+        managed_instance_name,
+        resource_group_name,
+        retention_days,
+        deletion_date=None):
+    '''
+    Updates short term retention for database
+
+    Custom function makes create mode more convenient.
+    '''
+
+    if deletion_date:
+        database_name = '{},{}'.format(
+            database_name,
+            _to_filetimeutc(deletion_date))
+
+        client = \
+            get_sql_restorable_dropped_database_managed_backup_short_term_retention_policies_operations(
+                cmd.cli_ctx,
+                None)
+
+        policy = client.create_or_update(
+            restorable_dropped_database_id=database_name,
+            managed_instance_name=managed_instance_name,
+            resource_group_name=resource_group_name,
+            retention_days=retention_days)
+    else:
+        policy = client.create_or_update(
+            database_name=database_name,
+            managed_instance_name=managed_instance_name,
+            resource_group_name=resource_group_name,
+            retention_days=retention_days)
+
+    return policy
+
+
+def get_short_term_retention_mi(
+        cmd,
+        client,
+        database_name,
+        managed_instance_name,
+        resource_group_name,
+        deletion_date=None):
+    '''
+    Gets short term retention for database
+
+    Custom function makes create mode more convenient.
+    '''
+
+    if deletion_date:
+        database_name = '{},{}'.format(
+            database_name,
+            _to_filetimeutc(deletion_date))
+
+        client = \
+            get_sql_restorable_dropped_database_managed_backup_short_term_retention_policies_operations(
+                cmd.cli_ctx,
+                None)
+
+        policy = client.get(
+            restorable_dropped_database_id=database_name,
+            managed_instance_name=managed_instance_name,
+            resource_group_name=resource_group_name)
+    else:
+        policy = client.get(
```

Comment:
If deletion date is given, you will return dropped database related information and if not, you will return database information directly, right?
Can live database and dropped database be the same name? If yes, in your current scenario, as soon as deletion date is given, you will only show up the dropped database information, right?

Reply:
Correct, there can exist live and multiple dropped databases with the same name. If deletion date is provided I will use client for dropped databases and fetch them, otherwise I will use client for live databases.